beacon-chain/rpc/eth/events/events.go

```diff
@@ -232,6 +248,83 @@ func handleStateEvents(
 	}
 }
 
+// notifyPayloadAttributesStream on successful FCU notify the event stream that a payload was sent
+func (s *Server) streamPayloadAttributes(stream ethpbservice.Events_StreamEventsServer) error {
+	headState, err := s.HeadFetcher.HeadStateReadOnly(s.Ctx)
+	if err != nil {
+		return err
+	}
+
+	headBlock, err := s.HeadFetcher.HeadBlock(s.Ctx)
+	if err != nil {
+		return err
+	}
+
+	headRoot, err := s.HeadFetcher.HeadRoot(s.Ctx)
+	if err != nil {
+
+		return err
+	}
+
+	headPayload, err := headBlock.Block().Body().Execution()
+	if err != nil {
+
+		return err
+	}
+
+	t, err := slots.ToTime(uint64(headState.GenesisTime()), headState.Slot())
+	if err != nil {
+		return err
+	}
+
+	prevRando, err := helpers.RandaoMix(headState, time.CurrentEpoch(headState))
+	if err != nil {
+		return err
+	}
+
+	switch headState.Version() {
+	case version.Bellatrix:
+		return streamData(stream, PayloadAttributesTopic, &ethpb.EventPayloadAttributeV1{
+			Version: version.String(headState.Version()),
+			Data: &ethpb.EventPayloadAttributeV1_BasePayloadAttribute{
+				ProposerIndex:     headBlock.Block().ProposerIndex(),
```

Comment:
I think this should be the proposer index of `headState.slot + 1`. We want to advance the state by 1 and use `BeaconProposerIndex`

Reply:
I checked the API again. This is what it says
```
 - `proposal_slot`: the slot at which a block using these payload attributes may be built.
```

Based on that, I'm taking it as `headState.slot`. Your current implementation is right